### PR TITLE
fs/mnemofs: Refactoring path logic, direntry size bug fix, open free bug fix

### PR DESCRIPTION
**Pull request by resyfer**
## Summary

- The entire path logic is being refactored, and rewritten to ensure no logical flaws.
- Direntry size bug is fixed which prevented proper iteration of the directories.
- Open free bug fixed to prevent memory leak after close.

## Impact
mnemofs will partly have correct path logic and `ls` works flawlessly. Also `open()` and `close()` work perfectly.

## Testing
checkpatch and NAND flash virtual device.

### CODE REVIEW

**Comment by resyfer** [comment]
@acassis Should I make this as a draft PR?

File: fs/mnemofs/mnemofs.c [hunk 1]
```diff
@@ -1183,7 +1183,11 @@ static int mnemofs_opendir(FAR struct inode *mountpt,
       goto errout_with_pitr;
     }
 
-  mfs_pitr_init(sb, path, depth, pitr, false);
+  ret = mfs_pitr_init(sb, path, depth, pitr, true);
+  if (predict_false(ret < 0))
+    {
+      goto errout_with_fsdirent;
```
Comment (acassis) [inline fs/mnemofs/mnemofs.c:1189]: Could be a good idea to include a debug error msg here

File: fs/mnemofs/mnemofs.c [hunk 2]
```diff
@@ -1787,7 +1800,12 @@ static int mnemofs_mkdir(FAR struct inode *mountpt, FAR const char *relpath,
   finfo("Lock acquired.");
 
   flags = mfs_get_patharr(sb, relpath, &path, &depth);
-  if ((flags & MFS_NEXIST) ==  0)
+  if ((flags & MFS_EXIST) != 0)
+    {
+      ret = -EEXIST;
```
Comment (acassis) [inline fs/mnemofs/mnemofs.c:1805]: Please include debug error msg here too

File: fs/mnemofs/mnemofs_blkalloc.c [hunk 3]
```diff
@@ -503,29 +503,45 @@ void mfs_ba_pgmarkdel(FAR struct mfs_sb_s * const sb, mfs_t pg)
...
-  /* TODO */
+  int   ret = OK;
+  mfs_t i;
+
+  for (i = 1; i < MFS_NBLKS(sb); i++)
+    {
+      if (DEL_ARR_BLK(sb, i) == MFS_PGINBLK(sb))
+        {
+          ret = mfs_erase_blk(sb, i);
+
```
Comment (acassis) [inline fs/mnemofs/mnemofs_blkalloc.c:519]: remove extra empty line

**Review by acassis — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**